src/process.cpp

```diff
@@ -233,47 +238,16 @@ std::vector<ctx_t> &proc_t::get_apps() {
 // Returns default image if image configuration is not set.
 // Returns http content-type header compatible image type.
 std::string proc_t::get_app_image(int app_id) {
-  auto default_image = SUNSHINE_ASSETS_DIR "/box.png";
-
   auto iter           = std::find_if(_apps.begin(), _apps.end(), [&app_id](const auto app) {
     return app.id == std::to_string(app_id);
   });
   auto app_image_path = iter == _apps.end() ? std::string() : iter->image_path;
 
-  if(app_image_path.empty()) {
-    BOOST_LOG(warning) << "Couldn't find app image for ID ["sv << app_id << ']';
-    return default_image;
-  }
-
-  // get the image extension and convert it to lowercase
-  auto image_extension = std::filesystem::path(app_image_path).extension().string();
-  boost::to_lower(image_extension);
-
-  // return the default box image if extension is not "png"
-  if(image_extension != ".png") {
-    return default_image;
-  }
-
-  // check if image is in assets directory
-  auto full_image_path = std::filesystem::path(SUNSHINE_ASSETS_DIR) / app_image_path;
-  if(std::filesystem::exists(full_image_path)) {
-    return full_image_path.string();
-  }
-  else if(app_image_path == "./assets/steam.png") {
-    // handle old default steam image definition
-    return SUNSHINE_ASSETS_DIR "/steam.png";
-  }
-
-  // check if specified image exists
-  std::error_code code;
-  if(!std::filesystem::exists(app_image_path, code)) {
-    // return default box image if image does not exist
-    return default_image;
+  auto image_path = validate_app_image_path(app_image_path);
+  if(image_path == DEFAULT_APP_IMAGE_PATH) {
+    BOOST_LOG(info) << "Couldn't find non-default app image for ID ["sv << app_id << ']';
```

Comment:
We probably don't want to print a log message like this unless the user specified an image and we couldn't load it.